torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1885,6 +1967,13 @@ def _simplify_floor_div(self, expr):
             self.evaluate_expr(eq_expr)
         return self.simplify(expr)
 
+    def _verify_valid_range(self, symbol, valid_range):
+        if symbol not in self.var_to_val:
+            return
+        if has_hint(symbol) and self.size_hint(symbol) not in valid_range:
+            raise RuntimeError(f"Valid range, {valid_range}, contradicts "
+                               f"traced value of {symbol}, {self.size_hint(symbol)}")
```

Comment:
I want two things here:

1. I want this sanity check at the point where the user said constrain_range / setup the range in the symbol
2. I want a better message, something like "You specified that {source} should fall within {valid_range}, but the concrete value we are tracing with has size {self.size_hint(symbol)} which is inconsistent with the range." Importantly, you must not print the symbol, as this is a meaningless s0; you must print a source of the symbol (or maybe print all of them.)

Reply:
> I want this sanity check at the point where the user said constrain_range / setup the range in the symbol

That was there, but then you told me to move it. I can put it back. Which function is the best place for it?

> I want a better message, something like ... 

Sure thing.